netty/src/main/java/io/grpc/netty/WriteQueue.java

```diff
@@ -100,6 +100,18 @@ ChannelFuture enqueue(QueuedCommand command, ChannelPromise promise, boolean flu
   }
 
   /**
+   * Enqueue the runnable. It is not safe for another thread to queue an Runnable directly to the
+   * event loop, because it will be out-of-order with writes. This method allows the Runnable to be
+   * processed in-order with writes.
+   */
+  void enqueue(Runnable runnable, boolean flush) {
+    queue.add(new RunnableCommand(runnable));
```

Comment:
you might consider chaining this with the above enqueue and just use a voidPromise